src/main/java/org/kohsuke/github/GHRepositoryRule.java

```diff
@@ -0,0 +1,667 @@
+    }
+
+    /**
+     * Gets the ruleset source.
+     *
+     * @return the ruleset source
+     */
+    public String getRulesetSource() {
+        return this.rulesetSource;
+    }
+
+    /**
+     * Gets the ruleset id.
+     *
+     * @return the ruleset id
+     */
+    public long getRulesetId() {
+        return this.rulesetId;
+    }
+
+    /**
+     * Gets a parameter. ({@link GHRepositoryRule.Parameters Parameters} provides a list of available parameters.)
+     *
+     * @param parameter
+     *            the parameter
+     * @param <T>
+     *            the type of the parameter
+     * @return the parameters
+     */
+    public <T> Optional<T> getParameter(Parameter<T> parameter) {
+        return Optional.ofNullable(this.parameters).map(p -> p.get(parameter.getKey())).map(parameter);
+    }
+
+    /**
+     * The type of the ruleset.
+     */
+    public static enum Type {
+        /**
+         * unknown
+         */
+        UNKNOWN,
+
+        /**
+         * creation
+         */
+        CREATION,
+
+        /**
+         * update
+         */
+        UPDATE,
+
+        /**
+         * deletion
+         */
+        DELETION,
+
+        /**
+         * required_linear_history
+         */
+        REQUIRED_LINEAR_HISTORY,
+
+        /**
+         * required_deployments
+         */
+        REQUIRED_DEPLOYMENTS,
+
+        /**
+         * required_signatures
+         */
+        REQUIRED_SIGNATURES,
+
+        /**
+         * pull_request
+         */
+        PULL_REQUEST,
+
+        /**
+         * required_status_checks
+         */
+        REQUIRED_STATUS_CHECKS,
+
+        /**
+         * non_fast_forward
+         */
+        NON_FAST_FORWARD,
+
+        /**
+         * commit_message_pattern
+         */
+        COMMIT_MESSAGE_PATTERN,
+
+        /**
+         * commit_author_email_pattern
+         */
+        COMMIT_AUTHOR_EMAIL_PATTERN,
+
+        /**
+         * committer_email_pattern
+         */
+        COMMITTER_EMAIL_PATTERN,
+
+        /**
+         * branch_name_pattern
+         */
+        BRANCH_NAME_PATTERN,
+
+        /**
+         * tag_name_pattern
+         */
+        TAG_NAME_PATTERN,
+
+        /**
+         * workflows
+         */
+        WORKFLOWS,
+
+        /**
+         * code_scanning
+         */
+        CODE_SCANNING
+    }
+
+    /**
+     * The source of the ruleset type.
+     */
+    public enum RulesetSourceType {
+        /**
+         * unknown
+         */
+        UNKNOWN,
+
+        /**
+         * Repository
+         */
+        REPOSITORY,
+
+        /**
+         * Organization
+         */
+        ORGANIZATION
+    }
+
+    /**
+     * Available parameters for a ruleset.
+     */
+    public interface Parameters {
+        /**
+         * update_allows_fetch_and_merge parameter
+         */
+        public static final BooleanParameter UPDATE_ALLOWS_FETCH_AND_MERGE = new BooleanParameter(
+                "update_allows_fetch_and_merge");
```

Comment:
Which value of the `GHRepositoryRule.Type` enum  applies here?  Is it suppose to be `REQUIRED_LINEAR_HISTORY`? 

I'm fine with `Parameter<T>` taking string in the constructor, but  in `Parameters` I want those strings to be generated from `GHRepositoryRule.Type` to reduce the likelihood of the strings here getting out of sync with the enum values.

Reply:
I am not sure, what do do... Enum values cannot have static fields if you think about sth like `Type.REQUIRED_LINEAR_HISTORY.UPDATE_ALLOWS_FETCH_AND_MERGE`.

Reply:
@bitwiseman what should I do here?